File: apple/Sources/ReactNativeWatchOS/Modifiers/FixedSize.swift
import SwiftUI

/// SwiftUI `.fixedSize()` / `.fixedSize(horizontal:vertical:)`. With neither
/// `horizontal` nor `vertical` param present, fixes the view at its ideal
/// size on both axes; otherwise fixes only the requested axes (a missing
/// axis param defaults to `false`).
enum RNWFixedSizeModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("fixedSize") { view, params, _ in
            let h = params.bool("horizontal")
            let v = params.bool("vertical")
            if h == nil && v == nil {
                return AnyView(view.fixedSize())
            }
            return AnyView(view.fixedSize(
                horizontal: h ?? false,
                vertical: v ?? false
            ))
        }
    }
}

File: src/modifiers/fixedSize.ts
import { createModifier } from './createModifier';

export interface FixedSizeParams {
  /// Fix the ideal width, so the view never compresses or stretches
  /// horizontally.
  horizontal?: boolean;
  /// Fix the ideal height, so the view never compresses or stretches
  /// vertically.
  vertical?: boolean;
}

/// SwiftUI `.fixedSize()` / `.fixedSize(horizontal:vertical:)`. With no
/// params the view is fixed at its ideal size on both axes. Pass
/// `horizontal` / `vertical` booleans to fix only specific axes.
export function fixedSize(params?: FixedSizeParams) {
  return createModifier('fixedSize', params);
}
